pkg/controller/controller.go

```diff
@@ -60,4 +61,11 @@ func Add(mgr manager.Manager, options Options) {
 	if err := c.Watch(&source.Kind{Type: u}, &crthandler.EnqueueRequestForObject{}); err != nil {
 		log.Fatal(err)
 	}
+	r := NewReconcileLoop(time.Duration(time.Minute)*1, options.GVK, mgr.GetClient())
+	cs := &source.Channel{Source: r.Source}
+	cs.InjectStopChannel(r.Stop)
```

Comment:
From tracing through code and comments, it looks like the controller should do this for us. What happens if you just leave this out?

https://github.com/kubernetes-sigs/controller-runtime/blob/8d93cc75ed7086e85ec7de09332a3bfd8603a959/pkg/runtime/inject/inject.go#L111

I suspect we're not meant to provide this channel or try to use it ourselves.

Reply:
I think it wants to get this channel: https://github.com/kubernetes-sigs/controller-runtime/blob/8d93cc75ed7086e85ec7de09332a3bfd8603a959/pkg/manager/internal.go#L76

Reply:
So you shouldn't have to do anything extra, and your right that is what it is trying to use but https://github.com/kubernetes-sigs/controller-runtime/pull/148 needs to be merged to because it is injecting a nil channel.

Reply:
If we want to have a stop channel for our reconcile loop, I think we should get it in main.go from signals.SetupSignalHandler() and add it as a field in `github.com/water-hole/ansible-operator/pkg/controller.Options`